selectors/plugins.ts

```diff
@@ -41,6 +41,15 @@ export const getChannelHeaderPluginComponents = createSelector(
     },
 );
 
+export const getChannelHeaderMenuPluginComponents = createSelector(
+    'getChannelHeaderMenuPluginComponents',
+    (state: GlobalState) => state,
+    (state: GlobalState) => state.plugins.components.ChannelHeader,
+    (state, channelHeaderMenuComponents = []) => {
```

Comment:
Passing state into the result function of the selector (the final argument to `createSelector`) won't work here. The state will change pretty much every time we call this, and when it does, it'll cause the selector to recalculate and return a new array even if it looks identical to the previous one.

We might be able to fix that by breaking this into two smaller selectors though like:
1. The first selector would that takes the state and the ChannelHeader components and return a corresponding array of whether or not `shouldRender` returned true. It'll basically look the same as what you have now, but with a `map` so that we're returning an array of booleans instead of a filtered array of components.
2. Then, have a second selector that uses `createShallowSelector` (from `mattermost-redux/utils/helpers`) instead of `createSelector`, takes the input from the first selector, and does the filtering. Because it's using `createShallowSelector`, it'll be looking at if the array of booleans actually changed before deciding whether or not to recalculate and return a new result.

Broken up that way, `shouldRender` for each component will still be called on every state change as we want, but it'll only cause a new result to be returned if the result of any of the `shouldRender` functions actually changed.

Hopefully that makes at least some sense. If not, I can try to explain it a bit better.

Reply:
Also, would you mind adding some tests for this selector? Since it's a bit more complicated than these other ones which just return a value directly from the state, I think it could use some

Reply:
Ah, this is an interesting point you bring up. Makes perfect sense, let me go ahead and update with the suggested fix. Will also add some tests.

Reply:
@hmhealey Let me know if the updated code is along the lines of what you suggested. Implemented the two selectors as you described, with the first returning array of booleans which is then fed into the second selector to actually filter the selectors.

Reply:
Also introduced some tests as suggested.